modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQueryBuilder.java

```diff
@@ -359,6 +359,11 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryShardContext) throws I
                     "indexed document [{}/{}/{}] couldn't be found", indexedDocumentIndex, indexedDocumentType, indexedDocumentId
             );
         }
+        if(getResponse.isSourceEmpty()) {
+            throw new ElasticsearchException(
```

Comment:
@martijnvg Sounds good. What about if the doc does not exist? Currently it throws a `ResourceNotFoundException` which means the search response status code is a 404. Not ideal since a 404 status also means the searched index does not exist. If its OK with you I'll change it throw a `IllegalArgumentException` as well. This will match the behavior in `GeoShapeQueryBuilder` so if the doc and/or `_source` is missing an `IllegalArgumentException` will be thrown (400 status code).